File: src/fibonacci.js
function fibonacci(n) {
  // ...
  return 0;
};

[
  [[0], 0],
  [[1], 1],
  [[2], 1],
  [[3], 2],
  [[4], 3],
  [[5], 5],
  [[6], 8],
  [[7], 13],
  [[8], 21],
  [[9], 34],
  [[10], 55],
].forEach((scenario) => {
  const inputs = scenario[0];
  const want = scenario[1];
  const got = fibonacci(...inputs);
  console.assert(want === got, `with ${inputs}: ${want} === ${got}`);
})

File: src/fibonacci.lua
local function fibonacci(n)
  -- ...
  return 0
end

local tdt = {
  { 0,  0 },
  { 1,  1 },
  { 2,  1 },
  { 3,  2 },
  { 4,  3 },
  { 5,  5 },
  { 6,  8 },
  { 7,  13 },
  { 8,  21 },
  { 9,  34 },
  { 10, 55 },
}

for _, scenario in ipairs(tdt) do
  local input = scenario[1]
  local want = scenario[2]
  local got = fibonacci(input)
  assert(got == want, "with " .. input .. ": " .. want .. " == " .. got);
end
